src/pages/issues.tsx

```diff
@@ -9,36 +8,88 @@ import {
 } from '@/components/constants/messages';
 import { ISSUES_URL } from '@/components/constants/url';
+import axios from 'axios';
+
+type SearchFieldProps = {
+    searchText: string;
+    onSearchTextChanged: (event: any) => void;
+    onSearchTextSubmitted: () => void;
+    loading: boolean;
+};
+
+const SearchField = ({
+    searchText,
+    onSearchTextChanged,
+    onSearchTextSubmitted,
+    loading,
+}: SearchFieldProps) => {
+    return (
+        <div className={classNames.searchFieldContainer}>
+            <input
+                placeholder="Enter query string to search issues"
+                value={searchText}
+                onChange={onSearchTextChanged}
+                className={classNames.issueSearchInput}
+            />
+            <button
+                onClick={onSearchTextSubmitted}
+                className={classNames.issuesSearchSubmitButton}
+                disabled={loading || !searchText.trim()}
+            >
+                Submit
+            </button>
+        </div>
+    );
+};
 
 const Issues: FC = () => {
     const [issueList, setIssueList] = useState<[]>([]);
+    const [searchText, setSearchText] = useState('');
+    const [isLoading, setIsLoading] = useState(true);
+    const [error, setError] = useState<null | any>(null);
 
-    const { response, error, isLoading } = useFetch(ISSUES_URL, {
-        data: {
-            userData: {
-                roles: {
-                    super_user: true,
-                },
-            },
-        },
-    });
+    const fetchIssues = async () => {
+        try {
+            setIsLoading(true);
+            const res = await fetch(`${ISSUES_URL}?q=${searchText}`);
+            const data = await res.json();
+            if ('issues' in data) {
+                const issuesAndPullRequests: any = data.issues;
+                const onlyIssues = issuesAndPullRequests.filter(
+                    (issue: { hasOwnProperty: (arg0: string) => any }) =>
+                        !Object.prototype.hasOwnProperty.call(
+                            issue,
+                            'pull_request'
+                        )
+                );
+                setIssueList(onlyIssues);
+            }
+            setIsLoading(false);
+            setError(null);
+        } catch (error) {
+            setIssueList([]);
+            setError(error);
+            setIsLoading(false);
+        }
+    };
 
     useEffect(() => {
-        if ('issues' in response) {
-            const issuesAndPullRequests = response.issues;
-            const onlyIssues = issuesAndPullRequests.filter(
-                (issue: { hasOwnProperty: (arg0: string) => any }) =>
-                    !Object.prototype.hasOwnProperty.call(issue, 'pull_request')
-            );
-            setIssueList(onlyIssues);
-        }
-    }, [isLoading, response]);
+        fetchIssues();
+    }, []);
+
+    const onSearchTextChanged = (e: any) => setSearchText(e.target.value);
```

Comment:
any ?

Reply:
Updated to `Reach.ChangeEvent<HTMLInputElement>`